src/widgets/unfold_widgets/preprocessing_widget.py

```diff
@@ -156,17 +157,48 @@ def add_columns_to_layout(self):
         columns = self.engine.get_columns()
         for column in columns:
             checkbox = QCheckBox(column)
-            checkbox.setChecked(True)
+            checkbox.setChecked(False)
             self.columns_group_form_layout.addRow(checkbox)
 
     def clear_column_layout(self):
         for i in reversed(range(self.columns_group_form_layout.count())):
             self.columns_group_form_layout.itemAt(i).widget().setParent(None)
 
     def submit_columns(self):
+        self.data_submitted = False
         columns = []
         for i in range(self.columns_group_form_layout.count()):
             if self.columns_group_form_layout.itemAt(i).widget().isChecked():
                 columns.append(self.columns_group_form_layout.itemAt(i).widget().text())
-        self.engine.set_state(columns)
-        self.get_data()
+        if not columns:
+            error = QMessageBox()
+            error.setIcon(QMessageBox.Critical)
+            error.setText('No columns were chosen')
+            error.setWindowTitle("Error")
+            error.exec_()
+            return
+        if self.engine.has_rows_with_nulls():
+            self.remove_nulls_warning()
+        if self.data_submitted:
+            self.engine.set_state(columns)
+            self.engine.clean_data("remove")
+            self.get_data()
+
+    def remove_nulls_warning(self):
+        warning = QMessageBox()
+        warning.setIcon(QMessageBox.Warning)
+        warning.setText("Null values in set")
+        warning.setInformativeText("Some rows will be deleted cause of the null values. You can check your data or "
+                                   "estimate it. Or you might go further, letting app remove some rows automatically. "
+                                   "Continue?")
```

Comment:
"This data contains some empty values. After proceeding some of the rows will be discarded. Continue?"